src/nifake/unit_tests/test_session.py

```diff
@@ -169,19 +173,19 @@ def test_session_context_manager_close_with_error(self):
     def test_lock_session_none(self):
         with nifake.Session('dev1') as session:
             session.lock()
-            self.patched_library.niFake_LockSession.assert_called_once_with(_matchers.ViSessionMatcher(SESSION_NUM_FOR_TEST), None)
+            self.patched_library.niFake_LockSession.assert_called_once_with(_matchers.ViSessionMatcher(SESSION_NUM_FOR_TEST), _matchers.ViBooleanPointerMatcher())
```

Comment:
Match any Boolean because caller_has_lock is a Boolean, right?

Reply:
No, it's any Boolean _pointer_, because the now-generic `lock` function in LibraryInterpreter returns the boolean, but it's ignored at the Session layer.

(When `lock` was implement purely as a session.py fancy function, we passed None for the boolean pointer, because that output is ignored.)